homeassistant/components/switch/wemo.py

```diff
@@ -109,23 +120,88 @@ def device_state_attributes(self):
 
         if self.insight_params or (self.coffeemaker_mode is not None):
             attr[ATTR_CURRENT_STATE_DETAIL] = self.detail_state
+            attr[ATTR_POWER_CURRENT_W] = self.power_current_watt
+            # attr[ATTR_POWER_AVG_W] = self.power_average_watt
+            attr[ATTR_POWER_TODAY_MW_MIN] = self.power_today_mw_min
+            attr[ATTR_POWER_TOTAL_MW_MIN] = self.power_total_mw_min
+            attr[ATTR_ON_FOR_TIME] = self.on_for
+            attr[ATTR_ON_TODAY_TIME] = self.on_today
+            attr[ATTR_ON_TOTAL_TIME] = self.on_total
+            attr[ATTR_POWER_THRESHOLD] = self.power_threshold
 
         if self.coffeemaker_mode is not None:
             attr[ATTR_COFFEMAKER_MODE] = self.coffeemaker_mode
 
         return attr
 
+#    @property
+    def _current_power_mw(self):
+        """Current power usage in mW."""
+        if self.insight_params:
+            try:
+                return self.insight_params['currentpower']
+            except KeyError:
+                return None
+
+    @property
+    def power_current_watt(self):
+        """Current power usage in W."""
+        if self.insight_params:
+            try:
+                return self._current_power_mw() / 1000
+            except TypeError:
+                return None
+
+    @property
```

Comment:
Why are somethings properties and other methods ? Also, why are they properties/method to begin with and not just all part of `device_state_attributes` ?